app/models/employer_events/employer_importer.rb

```diff
@@ -42,6 +42,26 @@ def plan_year_values
       end
     end
 
+    def issuer_ids 
+      @xml.xpath("//cv:benefit_groups", XML_NS).map do |node|
+        node.xpath("//cv:benefit_group/cv:elected_plans", XML_NS).map do |inner_node|
+          ids = inner_node.xpath("//cv:elected_plan/cv:carrier/cv:id/cv:id", XML_NS).map do |id|
+            stripped_node_value(id)
+          end
+         return ids.flatten || []
+        end
+      end
+    end
+
+    def create_plan_year(pyvs, employer_id)
+      if issuer_ids.present?
+        pyvs = (pyvs.merge(:employer_id => employer_id, :issuer_ids => issuer_ids))
+      else 
+        pyvs = (pyvs.merge(:employer_id => employer_id))
+      end
+      PlanYear.create!(pyvs)
+    end
+
```

Comment:
Several issues here, and I'll add them under my notes for 40007 as well.  Remember the ID you are pulling from the XML is the HBX ID of the employer in GlueDB.  The Employer ID in the plan year is the ObjectID of the GlueDB employer record.  Similarly, the Carrier ID is the HBX ID of the carrier as it is in glue, and this attribute needs to be set to the ObjectIDs of the corresponding carrier records.

Reply:
assigning mongo ids of carriers instead of hbx_carrier ids

Reply:
also added an update PY method in case a matching PY is found